### PR TITLE
ESP32: Deprecate make support

### LINKED ISSUE #6850: ESP32: Remove the support of make for examples
#### Problem
Currently, ESP32 examples in CHIP support both automake and CMake. As the primary build system was automake, people generally use it for compilation and the examples often break with CMake.

As ESP-IDF, also supports CMake as the default since v4.0, we should instead keep CMake as the only build system.

 #### Proposed Solution
Remove/deprecate automake support for ESP32 examples

### PR DESCRIPTION
**Pull request by sweetymhaiske**
#### Problem
* Currently ESP32 supports both CMake and Make build systems.
* Developer need to update build files for both build systems, which is in a way tedious.
* ESP-IDF is highly CMake compliant, hence to enhance performance by focusing on a single build system Make support is being deprecated.
* Fixes #6850 

#### Change overview

1. Removed below files from all esp32 examples
    - Makefile
    - component.mk
    - idf.sh
2. Made needed changes for successful CI.

#### Testing
* Compiled all esp32 examples using CMake (idf.py)
* Successful CI

### CODE REVIEW

**Comment by sweetymhaiske** [comment]
Github CI fails with this error https://github.com/project-chip/connectedhomeip/pull/7034/checks?check_run_id=2639671738#step:4:504. How can it be resolved?

**Comment by andy31415** [comment]
> Github CI fails with this error https://github.com/project-chip/connectedhomeip/runs/2639767629?check_suite_focus=true#step:4:504 How can it be resolved?

I believe we need to update the docker image. Discussed with @dhrishi - I believe updating the esp32 images should work (and tie them to a commit instead of a branch).

File: src/test_driver/esp32/qemu_setup.sh [hunk 1]
```diff
@@ -34,7 +34,6 @@ die() {
 cd "$here" || die 'ack!, where am I?!?'
 
 # shellcheck source=/dev/null
-source idf.sh
 rm -f ./build/sdkconfig
-SDKCONFIG=./build/sdkconfig SDKCONFIG_DEFAULTS=sdkconfig_qemu.defaults idf make defconfig
-SDKCONFIG=./build/sdkconfig idf make -j8 esp32_elf_builder
+SDKCONFIG=./build/sdkconfig SDKCONFIG_DEFAULTS=sdkconfig_qemu.defaults idf.py build defconfig
```
Comment (dhrishi) [inline src/test_driver/esp32/qemu_setup.sh:38]: This won't work. As this command is not valid for CMake `idf.py build defconfig`

File: src/test_driver/esp32/qemu_setup.sh [hunk 2]
```diff
@@ -34,7 +34,6 @@ die() {
...
+SDKCONFIG=./build/sdkconfig idf.py build esp32_elf_builder
```
Comment (dhrishi) [inline src/test_driver/esp32/qemu_setup.sh:39]: @andy31415 Is it possible to try this locally to confirm. If yes, are the steps captured anywhere that can help?

File: scripts/examples/esp_example.sh [hunk 3]
```diff
@@ -29,18 +30,26 @@ if [ -z "$app" ]; then
     exit 1
 fi
 
+git -C "$IDF_PATH" submodule update --init --recursive
+cd "$IDF_PATH"
+./install.sh
+. ./export.sh
+
+cd "$work_dir"
```
Comment (dhrishi) [inline scripts/examples/esp_example.sh:38]: Why was this not required previously and is required now? cc @andy31415

**Review by dhrishi — CHANGES_REQUESTED** [review]

**Comment by arunbharadwaj** [comment]
Is the README at https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/esp32 still accurate regarding the instructions on how to build esp32 image after this change? If not, can you please update it as well? Thanks.

**Comment by dhrishi** [comment]
>Is the README at https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/esp32 still accurate regarding the instructions on how to build esp32 image after this change? If not, can you please update it as well? Thanks.

@arunbharadwaj We had previously removed the mention of "idf make" from the README.md so that the new folks use CMake. But we can check if things need to be cleaned up there. Thanks! cc @sweetymhaiske

File: scripts/examples/esp_example.sh (hunk 3 above)
Comment (sweetymhaiske) [reply scripts/examples/esp_example.sh:38]: @drishi I have changed the current directory to $IDF_PATH for running `install.sh` and `export.sh`, hence I have added that.

Comment (dhrishi) [reply scripts/examples/esp_example.sh:38]: Ok. But, IMO, install.sh was happening before this as well. We need to check where does that come from

File: src/test_driver/esp32/qemu_setup.sh (hunk 1 above)
Comment (sweetymhaiske) [reply src/test_driver/esp32/qemu_setup.sh]: Yes, I checked it locally. It is removed.

File: src/test_driver/esp32/qemu_setup.sh (hunk 2 above)
Comment (sweetymhaiske) [reply src/test_driver/esp32/qemu_setup.sh]: For qemu, I have kept CI compilation as it is i.e using Make.

**Comment by sweetymhaiske** [comment]
> > Is the README at https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/esp32 still accurate regarding the instructions on how to build esp32 image after this change? If not, can you please update it as well? Thanks.
> 
> @arunbharadwaj We had previously removed the mention of "idf make" from the README.md so that the new folks use CMake. But we can check if things need to be cleaned up there. Thanks! cc @sweetymhaiske

README.md for esp32 examples demonstrate the CMake build system. I rechecked it.

File: scripts/examples/esp_example.sh (hunk 3 above)
Comment (sweetymhaiske) [reply scripts/examples/esp_example.sh]: Yes @dhrishi right. I have used export.sh only, `install.sh` happens in Dockerfile.

File: scripts/examples/esp_example.sh [hunk 4]
```diff
@@ -31,16 +32,21 @@ fi
 
 source "scripts/activate.sh"
 # shellcheck source=/dev/null
-source "$root"/idf.sh
+cd "$IDF_PATH"
+. ./export.sh
 
+cd "$work_dir"
```
Comment (dhrishi) [inline scripts/examples/esp_example.sh:38]: Can we use `cd -` instead of using a new variable?

**Review by dhrishi — CHANGES_REQUESTED** [review]

**Comment by dhrishi** [comment]
@sweetymhaiske  Why is `src/test_driver/esp32/idf.sh` in the changes, if you have not done any changes there?

**Comment by sweetymhaiske** [comment]
> @sweetymhaiske Why is `src/test_driver/esp32/idf.sh` in the changes, if you have not done any changes there?

**Comment by sweetymhaiske** [comment]
@dhrishi resolved. Thanks

**Comment by sweetymhaiske** [comment]
> @sweetymhaiske Why is `src/test_driver/esp32/idf.sh` in the changes, if you have not done any changes there?

@dhrishi resolved. Thanks

File: scripts/examples/esp_example.sh (hunk 4 above)
Comment (sweetymhaiske) [reply scripts/examples/esp_example.sh]: Done.

File: src/test_driver/esp32/qemu_setup.sh (hunk 2 above)
Comment (dhrishi) [reply src/test_driver/esp32/qemu_setup.sh]: @sweetymhaiske  Please file an isse for this so that it is tracked

**Comment by dhrishi** [comment]
LGTM. @andreilitvin  Can you please review this? @rgoliver This PR is ready now. So you should be unblocked.

File: src/test_driver/esp32/qemu_setup.sh (hunk 2 above)
Comment (sweetymhaiske) [reply src/test_driver/esp32/qemu_setup.sh]: #7186 Here it is.

**Review by dhrishi — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: scripts/examples/esp_example.sh [hunk 5]
```diff
@@ -31,16 +31,21 @@ fi
 
 source "scripts/activate.sh"
 # shellcheck source=/dev/null
-source "$root"/idf.sh
+cd "$IDF_PATH"
```
[the remark this answers is not in the document]
Comment (dhrishi) [reply scripts/examples/esp_example.sh:34]: It should work, IMO

[the remark this answers is not in the document]
Comment (sweetymhaiske) [reply scripts/examples/esp_example.sh]: Done.

File: scripts/examples/esp_example.sh [hunk 6]
```diff
@@ -31,16 +31,21 @@ fi
...
+. ./export.sh
 
+cd -
 for sdkconfig in "$root"/sdkconfig*.defaults; do
     # remove root path to get sdkconfig*.defaults name
     sdkconfig_name=${sdkconfig#"$root"/}
     rm -f "$root"/sdkconfig
-    SDKCONFIG_DEFAULTS=$sdkconfig_name idf make -j8 -C "$root" defconfig "$@"
-    idf make -j8 -C "$root" "$@" || {
+    cd "$root"
```
[the remark this answers is not in the document]
Comment (sweetymhaiske) [reply scripts/examples/esp_example.sh]: @andy31415 it works. I have made the needed changes.

**Review by msandstedt — APPROVED** [review]

**Comment by andy31415** [comment]
@sweetymhaiske - merge conflicts - should be able to merge once those are fixed.

**Comment by sweetymhaiske** [comment]
> @sweetymhaiske - merge conflicts - should be able to merge once those are fixed.

@andy31415 Resolved.

**Comment by dhrishi** [comment]
@sweetymhaiske  There seems to be a merge conflict again.

**Review by woody-apple — APPROVED** [review]

**Comment by sweetymhaiske** [comment]
> @sweetymhaiske There seems to be a merge conflict again.

Resolved.

**Pull request merged**